Submitted By: Robert Connolly <robert at linuxfromscratch dot org> (ashes)
Date: 2005-02-15
Initial Package Version: 0.9.27
Upstream Status: From Upstream
Origin: http://bugs.uclibc.org/view.php?id=15
Description: This patch builds ldconfig(8) dynamically linked.

#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- uClibc/utils/Makefile~dyn-ldconfig
+++ uClibc/utils/Makefile
@@ -45,7 +45,7 @@
 	$(STRIPTOOL) -x -R .note -R .comment $@
 
 ldconfig:	ldconfig.c
-	$(CC) $(CFLAGS) -Wl,-s -static \
+	$(CC) $(CFLAGS) -Wl,-s $(if $(filter $(UCLIBC_STATIC_LDCONFIG),y),-static) \
 		-DUCLIBC_RUNTIME_PREFIX=\"$(RUNTIME_PREFIX)\" \
 		-DUCLIBC_LDSO=$(UCLIBC_LDSO) -I. -I../ldso/include \
 		$^ -o $@
--- uClibc/extra/Configs/Config.in~dyn-ldconfig
+++ uClibc/extra/Configs/Config.in
@@ -311,6 +311,15 @@
 	  if you are using an older Linux kernel (2.0.x) that lacks large file 
 	  support.  Enabling this option will increase the size of uClibc.
 
+config UCLIBC_STATIC_LDCONFIG
+	bool "Link ldconfig statically"
+	default n
+	help
+	  If you wish to link the ldconfig binary statically, enable this option.
+	  It being static can be beneficial if you have a library problem and need
+	  ldconfig to recover from it.  If keeping the size down is more important
+	  to you than system recovery, disable this option.
+
 choice
 	prompt "Malloc Implementation"
 	default MALLOC if ! ARCH_HAS_MMU
